lib/multitransport/worker.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+const pino = require('../../pino.js')
+const build = require('pino-abstract-transport')
+
+module.exports = async function ({ transports }) {
+  transports = await Promise.all(transports.map(async (t) => {
```

Comment:
what about error handling?

suggest `Promise.allSettled` so that multiple failures can be surfaced rather than the first to error

Reply:
wouldn't this mean that long pending errors hold the rest in memory? (probably fine)

Reply:
I tend to side with @bmeck here.

Reply:
in that it's probably fine or in that you don't want to long pending errors to rest in memory?

it's a tension between devex and memory usage then, I guess it depends how bad the memory usage can get vs the value of someone seeing multiple errors in one run rather than fixing one issue to find another etc (especially given that these sorts of things may not be easily reproduced, and more errors in production would be valuable)

Reply:
I think the output for multiple errors would be really confusing. We `throw` the first, so it gets logged to stderr via `console.error` later on.

Reply:
you can throw a single `AggregateError` instead - since it's a native Error constructor familiarity with multiple errors as standardized through `AggregateError` will only increase

Reply:
> you can throw a single `AggregateError` instead - since it's a native Error constructor familiarity with multiple errors as standardized through `AggregateError` will only increase

I had no idea this exists. According to https://node.green it is only available from Node 15.14 and above.

Reply:
it can be polyfilled https://www.npmjs.com/package/es-aggregate-error

Reply:
let's punt this to a next iteration for now, if demand for better error handling becomes apparent I think that's the way to do it.